### ISSUE TITLE
MVS explorer changing editor syntax causes a callstack limit exception 

### ISSUE THREAD

**Issue by jordanCain** [zowe/zlux#369]
Since migrating from react 15 to 16 we're seeing the below error when changing syntax

```
Modal.js:193 Uncaught RangeError: Maximum call stack size exceeded.
    at HTMLDocument.Modal._this.enforceFocus (Modal.js:193)
    at DOMSelection._setDOMSelection (built-codeEdit.js:5358)
    at DOMSelection.setSelection (built-codeEdit.js:5357)
    at TextView._updateDOMSelection (built-codeEdit.js:7162)
    at TextView._handleFocus (built-codeEdit.js:6553)
    at HTMLDivElement.handler (built-codeEdit.js:7008)
    at DOMSelection._setDOMSelection (built-codeEdit.js:5358)
    at DOMSelection.setSelection (built-codeEdit.js:5357)
    at TextView._updateDOMSelection (built-codeEdit.js:7162)
    at TextView._handleFocus (built-codeEdit.js:6553)
```

**Comment by pranav1698** [comment]
@jordanCain I believe that this is an issue with the `material-ui` module
And, it can be solved using this answer https://stackoverflow.com/questions/54133326/material-ui-uncaught-rangeerror-maximum-call-stack-size-exceeded
But, I am not able to test the changes because when I run `npm run dev` it gives me this error 
![Screenshot from 2020-02-08 00-57-12](https://user-images.githubusercontent.com/34754265/74059403-fdb8d300-4a0d-11ea-92c9-52e6ebea351e.png)
What can I do to solve this one?

**Comment by jordanCain** [comment]
@pranav1698 the dev env for the explorers requires a backend to talk to, you can configure the backend for local development by setting the host variable in the urlUtils.js file https://github.com/zowe/explorer-mvs/blob/master/WebContent/js/utilities/urlUtils.js#L11 this should be set to the host and port of a zowe api gateway. By default a zowe install uses port 7554 for the api gateway. when you load the page after setting this correctly you should be prompted for credentials. 

At this point you may run into browser CORS issues, depending on which browser you use take a look at https://stackoverflow.com/questions/29954037/why-is-an-options-request-sent-and-can-i-disable-it/42725358#42725358 for chrome, or firefox there's a plugin which is called CORS everywhere which I've used in the past

**Comment by pranav1698** [comment]
I have locally hosted zowe API-Gateway but getting the same error because the explorer is trying to call wrong login endpoint, and I am getting `no handler for the endpoint called by the app`
I have tried searching https://github.com/zowe/api-layer/wiki/Zowe-Authentication-and-Authorization-Service to get to the login endpoint, without any success
Do you know what is the login endpoint for the zowe api gateway?

**Comment by jordanCain** [comment]
Whilst the explorer apps go via the gateway they rely on different back end services which are registered with the api gateway, in this case its the zowe/data-sets server. The back end service then communicates with z/OSMF which must run on z/OS.

The login endpoint this app uses is https://host:gatewayport/api/v1/data-sets/login 

If you do not have access to a remote system it may be too difficult to setup a suitable development environment in this circumstance. @NakulManchanda Can you provide details of the z-trial or mastet the mainframe system that Zowe has so pranav could get a backend?

**Comment by pranav1698** [comment]
@jordanCain was able to host the explorer on my local machine
![Screenshot from 2020-02-11 20-07-28](https://user-images.githubusercontent.com/34754265/74246845-29d49c80-4d0b-11ea-8d75-0f8bbfc96e5a.png)
Had to disable the cors web security on google chrome to disable web security issues 
The CORS everywhere plugin was not working for the explorer
Now working, on replicating the issue

**Comment by skurnevich** [comment]
Hi Jordan, Pranav, I'm going to take this as it is prerequisite to proper fix of zowe/zlux#292

**Comment by NakulManchanda** [comment]
PR to pass extra prop to MenuProps={{ disableEnforceFocus: true }} resolved the issue.
